### PR TITLE
[Tosa] Update Sin/Cos operators legalization

### PR DESCRIPTION
**Pull request by Jerry-Ge**
- with the introduction of tosa.sin and tosa.cos ops
- update the legalization to do direct mapping

### CODE REVIEW

**Comment by Jerry-Ge** [comment]
hi @jpienaar, could you help review/merge this patch? thanks :)

**Comment by Jerry-Ge** [comment]
Hi @rdzhabarov, can you help review this?

File: tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc [hunk 1]
```diff
@@ -270,43 +270,28 @@ LogicalResult ConvertTFSignOp::matchAndRewrite(
 LogicalResult ConvertTFSinOp::matchAndRewrite(Operation* op,
                                               PatternRewriter& rewriter) const {
   auto tf_sin_op = cast<TF::SinOp>(op);
-  ShapedType output_type =
-      mlir::cast<ShapedType>(tf_sin_op.getResult().getType());
 
-  std::optional<Value> result =
-      convertSinOp(rewriter, op, tf_sin_op.getX(), output_type);
-  if (!result) return failure();
+  ShapedType output_type = dyn_cast<TensorType>(tf_sin_op.getResult().getType());
```
Comment (jpienaar) [inline tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc:274]: OOC why cast to TensorType and assign to ShapedType?

Comment (jpienaar) [reply tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc:274]: Also this doesn't seem possible: it would fail verification.

**Review by jpienaar — COMMENTED** [review]
Overall this looks good, small changes and should be good to go.

File: tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc (hunk 1 above)
Comment (Jerry-Ge) [reply tensorflow/compiler/mlir/tosa/transforms/legalize_tf.cc]: thanks Jacques! it should be bug, updated in the latest PS

**Review by jpienaar — APPROVED** [review]

File: tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc [hunk 2]
```diff
@@ -3309,38 +3309,26 @@ LogicalResult ConvertTFLSinOp::matchAndRewrite(
...
+                                       tfl_sin_op.getX());
 
-  rewriter.replaceOp(op, {result.value()});
   return success();
 }
 
 LogicalResult ConvertTFLCosOp::matchAndRewrite(
     Operation* op, PatternRewriter& rewriter) const {
   auto tfl_cos_op = cast<TFL::CosOp>(op);
   Value input = tfl_cos_op.getX();
```
Comment (jpienaar) [inline tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc:3323]: Missed it (but presubmit caught) that this one is unused.

Comment (Jerry-Ge) [reply tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc]: thanks! removed this in the latest patch set.

**Comment by keerthanakadiri** [comment]
Hi @jpienaar , Can you please review this PR? Thank you !

**Review by jpienaar — APPROVED** [review]

**Comment by Jerry-Ge** [comment]
Hi @jpienaar , the CI looks good. Can we submit this patch now? Thanks!

**Pull request merged**